rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -815,10 +815,10 @@ void SqliteStorage::read_metadata()
   if (database_->field_exists("topics", "offered_qos_profiles")) {
     if (database_->field_exists("topics", "type_description_hash")) {
       std::string query =
-        "SELECT messages.topic_id, name, type, serialization_format, COUNT(messages.id), "
+        "SELECT topics.id, name, type, serialization_format, COUNT(messages.id), "
         "MIN(messages.timestamp), MAX(messages.timestamp), offered_qos_profiles, "
-        "type_description_hash FROM messages JOIN topics on topics.id = messages.topic_id "
-        "GROUP BY topics.name;";
+        "type_description_hash FROM topics LEFT OUTER JOIN messages on "
```

Comment:
this is the important part to join the table, previously excludes the contents from `topics` table if id does not match.

with this query, we can have the contents including message count is 0 as following.

```bash
sqlite> SELECT topics.id, name, type, serialization_format, COUNT(messages.id), MIN(messages.timestamp), MAX(messages.timestamp) FROM topics LEFT OUTER JOIN messages on topics.id = messages.topic_id GROUP BY topics.name;
5|/events/write_split|rosbag2_interfaces/msg/WriteSplitEvent|cdr|0||
4|/greetings|example_interfaces/msg/String|cdr|0||
1|/parameter_events|rcl_interfaces/msg/ParameterEvent|cdr|0||
3|/rosout|rcl_interfaces/msg/Log|cdr|9|1719079568093052120|1719079568099102171
2|/some_topic|std_msgs/msg/Bool|cdr|1|1719079568100694158|1719079568100694158
```

i have confirmed that this actually works with reproducible procedure explained in https://github.com/ros2/rosbag2/issues/1719